src/main/java/sg/ncl/MainController.java

```diff
@@ -432,23 +437,40 @@ private String getTeamIdByName(String teamName) {
     //--------------------------Account Settings Page--------------------------
     @RequestMapping(value="/account_settings", method=RequestMethod.GET)
     public String accountDetails(Model model, HttpSession session) throws IOException {
-    	// TODO id should be some session variable?
 
     	String userId_uri = properties.getSioUsersUrl() + session.getAttribute("id");
+//        String userId_uri = properties.getSioUsersUrl() + "AAAAAAAAAAAAAAAAAAAA";
         HttpHeaders headers = new HttpHeaders();
         headers.set("Authorization", AUTHORIZATION_HEADER);
 
         HttpEntity<String> request = new HttpEntity<String>("parameters", headers);
-        ResponseEntity responseEntity = restTemplate.exchange(userId_uri, HttpMethod.GET, request, String.class);
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(userId_uri, HttpMethod.GET, request, String.class);
+        String responseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                logger.error("No such user: {}", session.getAttribute("id"));
+                MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
+                throw new RestClientException("[" + error.getExceptionName() + "] ");
+            } else {
+                User2 user2 = extractUserInfo(responseBody);
+                originalUser = user2;
+                model.addAttribute("editUser", user2);
+                return "account_settings";
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)